tendrl/ceph_integration/manager/crud.py

```diff
@@ -119,3 +123,14 @@ def delete(self, object_type, object_id):
             )
         else:
             raise NotImplementedError(object_type)
+
+    def sync_request_status(self, request):
+        count = 0
+        while count < 30:
+            gevent.sleep(10)
+            if request.state == "complete" and not request.error:
+                return request.status
+            if request.error:
+                raise RequestStateError(request.error_message)
+            count += 1
+        raise request.error_message("Request timed out")
```

Comment:
How does this work? does "request.error_message" raise an exception or it itself is an exception?